build/common/installer/scripts/tomlparser.rb

```diff
@@ -209,8 +210,9 @@ def populateSettingValuesFromConfigMap(parsedConfig)
 
     #Get Kubernetes Metadata setting
     begin
-      if !parsedConfig[:log_collection_settings][:metadata_collection].nil? && !parsedConfig[:log_collection_settings][:metadata_collection][:enabled].nil?
+      if !parsedConfig[:log_collection_settings][:metadata_collection].nil? && !parsedConfig[:log_collection_settings][:metadata_collection][:enabled].nil? && !parsedConfig[:log_collection_settings][:metadata_collection][:include_fields].nil?
         @logEnableKubernetesMetadata = parsedConfig[:log_collection_settings][:metadata_collection][:enabled]
+        @logKubernetesMetadataiIncludeFields = parsedConfig[:log_collection_settings][:metadata_collection][:include_fields]
```

Comment:
looks good however we might be missing the default scenario.

1) customer enables metadata collection and does not specify field -> default scenario where we collect all field
2) customer enables metadata collection and does specify field -> we use the include_fields property.

as of now only option 2 is tackled

Reply:
I think you are talking about the scenario where enabled is true but include_fields is commented out. We can add a scenario where enabled is true and parsedConfig[:log_collection_settings][:metadata_collection][:include_fields] is nil.